File: 1 лаба/Python/H.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.dist = x * x + y * y


n = int(input())
points = []

for _ in range(n):
    x, y = map(int, input().split())
    points.append(Point(x, y))

points.sort(key=lambda p: p.dist)

for p in points:
    print(p.x, p.y)

File: 1 лаба/C++/H.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

struct Point {
  long long x;
  long long y;
  long long dist;
};

bool comparePoints(const Point &a, const Point &b) {
  return a.dist < b.dist;
}

int main() {
  int n;
  if (!(cin >> n)) return 0;
  vector<Point> points(n);
  for (int i = 0; i < n; i++) {
    cin >> points[i].x >> points[i].y;
    points[i].dist = points[i].x * points[i].x + points[i].y * points[i].y;
  }
  sort(points.begin(), points.end(), comparePoints);
  for (int i = 0; i < n; i++) {
    cout << points[i].x << " " << points[i].y << "\n";
  }
  return 0;
}
